### PR TITLE
Cleanup tempfiles

### PR DESCRIPTION
**Pull request by zhijianli88**
we introuduce a new testing feature T.cleanup to remove temp directory
T.cleanup was introduced since go-1.14: https://golang.org/pkg/testing/#T.Cleanup

**What this PR does / why we need it**:

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

File: pkg/repo/repotest/server.go [hunk 1]
```diff
@@ -36,11 +37,13 @@ import (
 //
 // The caller is responsible for destroying the temp directory as well as stopping
 // the server.
-func NewTempServer(glob string) (*Server, error) {
+func NewTempServer(t *testing.T, glob string) (*Server, error) {
```
Comment (bacongobbler) [inline pkg/repo/repotest/server.go:40]: As this is a public function signature, we cannot change it due to backwards compatibility concerns as per our contributing guide (see CONTRIBUTING.md).

I would suggest instead introducing a new function called `NewTempServerWithCleanup(t *testing.T, glob string) (*Server, error) {}`, and add a comment above `NewTempServer` that this function has been deprecated in favour of `NewTempServerWithCleanup`. I'd also add an additional comment to replace `NewTempServer` with `NewTempServerWithCleanup` in Helm 4.

Thanks!

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/repo/repotest/server.go (hunk 1 above)
Comment (zhijianli88) [reply pkg/repo/repotest/server.go]: Yeap, great thanks for your suggestion.

File: pkg/repo/repotest/server.go [hunk 2]
```diff
@@ -29,13 +30,41 @@ import (
...
 
+// NewTempServerWithCleanup creates a server inside of a temp dir.
+//
+// If the passed in string is not "", it will be treated as a shell glob, and files
+// will be copied from that path to the server's docroot.
+//
+// The caller is responsible for stopping the server.
+// The temp dir will be removed by testing package automatically when test finished.
+func NewTempServerWithCleanup(t *testing.T, glob string) (*Server, error) {
+	tdir, err := ioutil.TempDir("", "helm-repotest-")
```
Comment (bacongobbler) [inline pkg/repo/repotest/server.go:41]: To make this code more DRY, why not replace this all with the following:

```
func NewTempServerWithCleanup(t *testing.T, glob string) (*Server, error) {
    s, err := NewTempServer(glob)
    t.Cleanup(func() { os.RemoveAll(s.docroot) })
}
```

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Review by bacongobbler — APPROVED** [review]
LGTM.

**Pull request merged**